cmake/RunTests.cmake

```diff
@@ -38,7 +38,10 @@ set(ENV{SYSTEM_NAME} ${SYSTEM_NAME})
 execute_process(
   COMMAND ${BUSTED_PRG} ${TEST_TAG} ${TEST_FILTER} -v -o ${BUSTED_OUTPUT_TYPE}
     --lua=${LUA_PRG} --lazy --helper=${TEST_DIR}/${TEST_TYPE}/preload.lua
-    --lpath=${BUILD_DIR}/?.lua --lpath=?.lua ${TEST_PATH}
+    --lpath=${BUILD_DIR}/?.lua
+    --lpath=${WORKING_DIR}/runtime/lua/?.lua
```

Comment:
This is a try at sharing code between tests/runtime/core. Maybe we will do it differently, I will have a followup PR for further discussion, after this one. 

Related: #8677  cc @KillTheMule

Reply:
I don't understand makefiles, so just a thought: Is `?` recursive, so it also catches `runtime/lua/plugin/init.lua`? Also, does the test runner have access to `vim.api`? If so, great boon to writing tests!

Reply:
yes it's recursive.

> Also, does the test runner have access to vim.api? If so, great boon to writing tests!

`vim.api` is built-into Nvim so that would not be "shared". The test utilities could spawn `nvim` to use its features. We probably should do that, to reduce some of the code in `*/helpers.lua`.